pkg/download/blob.go

```diff
@@ -121,25 +121,35 @@ func CreateAppendBlob(blobURI, blobSas string) (*storage.Blob, error) {
 		return nil, err
 	}
 
-	fileName := getFileName(blobURI)
+	fileName, blobPathError := getBlobPathAfterContainerName(blobURI, containerRef.Name)
 	if fileName == "" {
-		return nil, fmt.Errorf("cannot extract file name from URL: %q", blobURI)
+		return nil, errors.Wrapf(blobPathError, "cannot extract blob path name from URL: %q", blobURI)
 	}
 
 	blobref := containerRef.GetBlobReference(fileName)
```

Comment:
what would happen if no nested directories were specified inside the container, fileName might be empty in that case right?

Reply:
We take blob URL. If no nested directories, blob name would be the "fileName"